### PR TITLE
User-configurable cron expression for regular cache flushing #2086

### PR DESCRIPTION
**Pull request by LukasCesal**
### Description (*)
(https://github.com/OpenMage/magento-lts/issues/2086)

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [ ] Add yourself to contributors list

### CODE REVIEW

**Review by elidrissidev — CHANGES_REQUESTED** [review]

File: app/code/core/Mage/Core/etc/config.xml
Comment (LukasCesal) [reply app/code/core/Mage/Core/etc/config.xml]: Ok, I will revert it to original value. But just for my information, is there any reason to flush all cache daily?
In our experience once a week is enough (and it is done just for sure) and there was not such cron before 1.7.x at all.

Comment (ADDISON74) [reply app/code/core/Mage/Core/etc/config.xml]: I don't see any reason to flush all cache daily. It can be done manually from Backend when needed and one time per week should be fine.

Please note the original value was (30 3 * * 6). As I see the new change is (30 2 * * *) which runs the cron job every day of the week at 2:30 am.

Comment (ADDISON74) [reply app/code/core/Mage/Core/etc/config.xml]: All right then. His explanation in Backend is more than enough. The PR looks good to me.

Comment (LukasCesal) [reply app/code/core/Mage/Core/etc/config.xml]: Yes, original value is 30 2 * * *  since magento 1.7.x (i do not know the reason, maybe it can "solve" potentital problems with cache records that never expires).
There was not such cron before 1.7.x, we have no problems without that feature that on 1.6.x for many years.

I understand this remark that some people can be relying on it currently and that is the reason the default value should be left as it was. I am ok with that, we have changed the value in our production environment.

But a lot (or some people) probably do not know about that feature at all so they can register some problems under heavy load at a certain redis configuration as we were.
image

Comment (elidrissidev) [reply app/code/core/Mage/Core/etc/config.xml]: I see, our website traffic is very low at 2am so we never face issues.

Since the PR requires at min 2 maintainers approvals, I have no objection if someone else thinks its better to change it to weekly :)

**Review by elidrissidev — APPROVED** [review]
Tested and it looks good from my side.

File: app/code/core/Mage/Core/etc/system.xml
Comment (kiatng) [inline app/code/core/Mage/Core/etc/system.xml:80]: Just curious, why do you recommend to run weekly at 03:30 on Saturday instead of the current default of daily at 02:30?

Suggest to enclose the value with double quotes: "0 0 30 2 0", "30 3 * * 6".
```suggestion
                            <comment>Use "0 0 30 2 0" to never run cron, the recommended setting is "30 3 * * 6"</comment>

Comment (elidrissidev) [reply app/code/core/Mage/Core/etc/system.xml:80]: It's mentioned here: https://github.com/OpenMage/magento-lts/pull/2087#discussion_r873553649

**Comment by Sekiphp** [comment]
For me is ok. Best option will be to merge this PR with original default value and create PR to 20.x for changed expression for weekly run.

**Review by Sekiphp — APPROVED** [review]

**Comment by ADDISON74** [comment]
I analyzed OpenMage Backend and I found only one section where we can configure a cron job. Below is the implementation of the Magento team for Sitemap.

google_sitemap

This PR can be modified to provide the same timing and we would keep a standard format of the change in Backend. I would choose two values: **Start Time** with 3 drop-downs (hour, minute, second) and **Frequency** which can be a drop-down with the same options like in the image (Always, Hourly, Daily, Weekly, Monthly, Yearly, Never). 

This is a more elegant and intuitive approach than filling in space delimited values in an input box with a help label bellow. In this way errors are avoided in entering wrong values that can affect the correct functionality.

**Comment by fballiano** [comment]
@ADDISON74 it's used somewhere like here:

Schermata 2022-05-19 alle 22 47 26

so to me it's really fine the way @LukasCesal built it :-)

**Review by luigifab — COMMENTED** [review]
A translate attribute is missing for `cache` tag: `translate="label"`.
A translate attribute is missing for `flush_cron_expr` tag: `translate="label comment"`.
Magento is now OpenMage, so instead of change that, I suggest to remove the word.
To never run cron, you just have to leave the field empty, right? I don't remember.

**Pull request merged**
